alf/environments/suite_go.py

```diff
@@ -232,12 +262,60 @@ def calc_area(self, board_indices=None):
             surrounded = self._board[B, y, x] == 0
             for dy, dx in self._dydxs:
                 neighbor = self._board[B, y + dy, x + dx]
-                surrounded = logical_and(surrounded, neighbor != -player)
-                surrounded = logical_and(surrounded, neighbor != 0)
-            area.append(logical_or(occupied, surrounded).sum(dim=(1, 2)))
+                surrounded = surrounded & (neighbor != -player)
+                surrounded = surrounded & (neighbor != 0)
+            area.append((occupied | surrounded).sum(dim=(1, 2)))
 
         return area[0], area[1]
 
+    def calc_area(self, board_indices=None):
+        """Calculate the area of each player.
+
+        In order for a position to be considered to be owned by a player, it has
+        to be either the player's stone or cannot be reached by the opponent's stones.
+        With this definition of area, players have to play until all the dead stones
+        have been taken out. This shouldn't change how the game is played. This
+        is the so called `Tromp-Taylor rules <https://en.wikibooks.org/wiki/Computer_Go/Tromp-Taylor_Rules>`_
+
```

Comment:
It would be very helpful if some more details regarding the algorithm implemented in line 295-309 can be added into the comments